ComponentKit/TransactionalDataSources/Mac/CKNSTableViewDataSource.mm

```diff
@@ -0,0 +1,193 @@
+{
+  self = [super init];
+  if (self) {
+    auto config = [[CKTransactionalComponentDataSourceConfiguration alloc] initWithComponentProvider:componentProvider
+                                                                                             context:context
+                                                                                           sizeRange:CKSizeRange()];
+
+    _componentDataSource = [[CKTransactionalComponentDataSource alloc] initWithConfiguration:config];
+
+    // Make sure we have 1 section
+    CKTransactionalComponentDataSourceChangeset *base =
+    [[CKTransactionalComponentDataSourceChangeset alloc] initWithUpdatedItems:nil
+                                                                 removedItems:nil
+                                                              removedSections:nil
+                                                                   movedItems:nil
+                                                             insertedSections:[NSIndexSet indexSetWithIndex:0]
+                                                                insertedItems:nil];
+
+    [_componentDataSource applyChangeset:base mode:CKTransactionalComponentDataSourceModeSynchronous userInfo:nil];
+
+    [_componentDataSource addListener:self];
+
+    _tableView = tableView;
+    _tableView.dataSource = self;
+    _tableView.delegate = self;
+    
+    _cellToItemMap = [NSMapTable weakToStrongObjectsMapTable];
+  }
+  return self;
+}
+
+- (instancetype)init
+{
+  CK_NOT_DESIGNATED_INITIALIZER();
+}
+
+#pragma mark - Changesets
+
+- (void)applyChangeset:(CKTransactionalComponentDataSourceChangeset *)changeset
+                  mode:(CKTransactionalComponentDataSourceMode)mode
+              userInfo:(NSDictionary *)userInfo
+{
+  [_componentDataSource applyChangeset:changeset mode:mode userInfo:userInfo];
+}
+
+- (id<NSObject>)modelForItemAtIndexPath:(NSIndexPath *)indexPath
+{
+  return [[[_componentDataSource state] objectAtIndexPath:indexPath] model];
+}
+
+- (CGSize)sizeForItemAtIndexPath:(NSIndexPath *)indexPath
+{
+  return [[[_componentDataSource state] objectAtIndexPath:indexPath] layout].size;
+}
+
+#pragma mark - Appearance announcement
+
+//- (void)announceWillAppearForItemInCell:(UICollectionViewCell *)cell
+//{
+//  _sendAppearanceEventForCell(cell, CKComponentAnnouncedEventTreeWillAppear, _cellToItemMap);
+//}
+//
+//- (void)announceDidDisappearForItemInCell:(UICollectionViewCell *)cell
+//{
+//  // We cannot use the indexPath directly, on deletion the indexPath of the deleted cell cannot be used to
+//  // get an item from the datasource.
+//  _sendAppearanceEventForCell(cell, CKComponentAnnouncedEventTreeDidDisappear, _cellToItemMap);
+//}
+//
+//NS_INLINE void _sendAppearanceEventForCell(UICollectionViewCell *cell, CKComponentAnnouncedEvent event, NSMapTable *cellToItemMap)
+//{
+//  [[[[cellToItemMap objectForKey:cell] lifecycleManager] scopeRoot] announceEventToControllers:event];
+//}
```

Comment:
Why not to delete this commented code? Unless you are going to make it work within this pull request. I'd defer this to further PR though.